torchaudio/backend/sox_io_backend.py

```diff
@@ -8,3 +10,72 @@
 def info(filepath: str) -> torch.classes.torchaudio.SignalInfo:
     """Get signal information of an audio file."""
     return torch.ops.torchaudio.sox_io_get_info(filepath)
+
+
+@_mod_utils.requires_module('torchaudio._torchaudio')
+def load(
+        filepath: str,
+        frame_offset: int = 0,
+        num_frames: int = -1,
+        normalize: bool = True,
+        channels_first: bool = True,
+) -> Tuple[torch.Tensor, int]:
+    """Load audio data from file.
+
+    This function can handle all the codecs that underlying libsox can handle, however note the
+    followings.
+
+    Note 1:
+        Current torchaudio's binary release only contains codecs for MP3, FLAC and OGG/VORBIS.
+        If you need other formats, you need to build torchaudio from source with libsox and
+        the corresponding codecs. Refer to README for this.
+
+    Note 2:
+        This function is tested on the following formats;
+         - WAV
+            - 32-bit floating-point
+            - 32-bit signed integer
+            - 16-bit signed integer
+            -  8-bit unsigned integer
+         - MP3
+         - FLAC
+         - OGG/VORBIS
+
+    By default, this function returns Tensor with ``float32`` dtype and the shape of ``[channel, time]``.
+    The samples are normalized to fit in the range of ``[-1.0, 1.0]``.
+
+    When the input format is WAV with integer type, such as 32-bit signed integer, 16-bit
+    signed integer and 8-bit unsigned integer (24-bit signed integer is not supported),
+    by providing ``normalize=False``, this function can return integer Tensor, where the samples
+    are expressed within the whole range of the corresponding dtype, that is, ``int32`` tensor
+    for 32-bit signed PCM, ``int16`` for 16-bit signed PCM and ``uint8`` for 8-bit unsigned PCM.
```

Comment:
I prefer a function that would always return a float between -1 and 1 when normalize=True, and a float unnormalized when False. Someone who wants the integer version would call a different function. Thoughts?

Reply:
That way, if user wants to get integer value, they have to go through float, which might cause numerical issue around the edge of value range without care. I do not think that's a good UX.

I/O is fundamental to scientific computing so loading function should provide the functionality to load data as is, without modifying it.
See also https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.wavfile.read.html

Reply:
Yes, it should definitely be possible to load the data as is, and without going through floats first. That being said, we could delegate that to a different function, and keep the main load function as returning floats. However, the default with `normalize=True` does just that, so I'm happy enough as it is.